### PR TITLE
Docs: Adds clarification notes to data proxy configuration

### PR DESCRIPTION
**Pull request by tomglenn**
**What this PR does / why we need it**:

Currently the plugin proxy does not support `urlParam` or `tokenAuth` configuration when used in the context of an App plugin. This PR adds clarification notes to the docs to reflect this.

### CODE REVIEW

**Review by academo — APPROVED** [review]

File: docusaurus/docs/how-to-guides/data-source-plugins/add-authentication-for-data-source-plugins.md [hunk 1]
```diff
@@ -253,6 +253,10 @@ Here's an example of adding `name` and `content` as URL parameters:
 ]
 ```
 
+:::note
+Be aware that `urlParams` configuration is only supported in Data source plugins. It is **NOT** supported in App plugins.
```
Comment (josmperez) [inline docusaurus/docs/how-to-guides/data-source-plugins/add-authentication-for-data-source-plugins.md:257]:
```suggestion
Be aware that `urlParams` configuration is only supported in data source plugins. It is _not_ supported in app plugins.
```

File: docusaurus/docs/how-to-guides/data-source-plugins/add-authentication-for-data-source-plugins.md [hunk 2]
```diff
@@ -297,6 +301,10 @@ Any parameters defined in `tokenAuth.params` are encoded as `application/x-www-f
 }
 ```
 
+:::note
+Be aware that `tokenAuth` configuration is only supported in Data source plugins. It is **NOT** supported in App plugins.
```
Comment (josmperez) [inline docusaurus/docs/how-to-guides/data-source-plugins/add-authentication-for-data-source-plugins.md:305]:
```suggestion
Be aware that `tokenAuth` configuration is only supported in data source plugins. It is _not_ supported in app plugins.
```

**Review by josmperez — APPROVED** [review]
Just a few minor style nits. LGTM.

**Comment by tomglenn** [comment]
Thanks @josmperez, fixed.

**Pull request merged**